### PR TITLE
Add hostname to metrics on panic

### LINKED ISSUE #937: Log hostname in the panic hook

### PR DESCRIPTION
**Pull request by sakridge**
Fixes #937

### CODE REVIEW

**Comment by sakridge** [comment]
Fixes https://github.com/solana-labs/solana/issues/937

**Comment by mvines** [comment]
Note: if you don't put "Fixes #937" in the PR description then #937 won't get auto closed.  Comments 1..N don't count

File: src/metrics.rs [hunk 1]
```diff
@@ -219,6 +220,13 @@ pub fn set_panic_hook(program: &'static str) {
                             None => "?".to_string(),
                         }),
                     )
+                    .add_field(
+                        "host",
+                        influxdb::Value::String(match hostname() {
```
Comment (mvines) [inline src/metrics.rs]: How about:
```rust
influxdb::Value::String(hostname().or(Ok("?".to_string())).unwrap())
```

Comment (sakridge) [reply src/metrics.rs]: yea a bit cleaner

**Pull request merged**